File: src/main/java/lkm/starterproject/accounting/entity/register/AccountTitle.java
package lkm.starterproject.accounting.entity.register;

import jakarta.persistence.*;
import lkm.starterproject.accounting.constants.UseStatus;
import lkm.starterproject.accounting.entity.company.Company;
import lkm.starterproject.auth.entity.BaseEntity;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter @Setter @ToString
@Builder @NoArgsConstructor @AllArgsConstructor
@Table(name = "account_title")
public class AccountTitle extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "account_title_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    private Long code;  //계정과목코드

    private String name;    // 계정명

    private String balanceClassification;   //대차구분

    private String type;    //계정종류

    @Builder.Default
    @OneToMany(mappedBy = "accountTitle", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Compendium> compendiums = new ArrayList<>();

    @Enumerated(EnumType.STRING)
    private UseStatus useStatus;    //사용구분

    private String properties;  //계정속성

    private String relatedWork; //관련업무

    private String cheque;   //수표

    private String balanceTallyClassification;  //잔액집계구분

    private String incomeExpenditureClassification; //수입지출구분

    private String incomeExpenditureParentAccount;  //수입지출 상위계정

    private String financialStatementName;  //재무제표 표시명

    private String financialStatementPrintingLocation;  //재무제표인쇄위치

    private String financialStatementAmountThickness;   //재무제표금액굵기

    private String financialStatementAmountParentheses; //재무제표금액괄호

    private String financialStatementHyperlinkTarget;   //재무제표 하이퍼링크대상

    private String financialStatementParentAccount; //재무제표  상위계정

}

File: vue/types/accounting/account-title.ts
import type {Compendium} from "~/types/accounting/compendium";

export interface AccountTitle {
    id: number;        //계정과목 코드
    code: number;
    name: string;    // 계정명
    balanceClassification: string;   //대차구분
    type: string;    //계정종류
    compendiums: Compendium;
    useStatus: string;    //사용구분
    properties: string;  //계정속성
    relatedWork: string; //관련업무
    cheque: string;   //수표
    balanceTallyClassification: string;  //잔액집계구분
    incomeExpenditureClassification: string; //수입지출구분
    incomeExpenditureParentAccount: string;  //수입지출 상위계정
    financialStatementName: string;  //재무제표 표시명
    financialStatementPrintingLocation: string;  //재무제표인쇄위치
    financialStatementAmountThickness: string;   //재무제표금액굵기
    financialStatementAmountParentheses: string; //재무제표금액괄호
    financialStatementHyperlinkTarget: string;   //재무제표 하이퍼링크대상
    financialStatementParentAccount: string; //재무제표  상위계정

}